### PR TITLE
Add a new field into J9HashTableState

### PR DESCRIPTION
**Pull request by hangshao0**
Add a new flag in J9HashTableState to control the behaviour when 
iterating the hash table.

Related to OpenJ9 PR https://github.com/eclipse/openj9/pull/10300

Signed-off-by: Hang Shao <hangshao@ca.ibm.com>

### CODE REVIEW

**Comment by hangshao0** [comment]
@gacholio, @dmitripivkine

File: include_core/omrhashtable.h [hunk 1]
```diff
@@ -65,6 +65,11 @@ extern "C" {
 #define J9HASH_TABLE_ITERATE_STATE_TREE_NODES 1
 #define J9HASH_TABLE_ITERATE_STATE_FINISHED  2
 
+/**
```
Comment (gacholio) [inline include_core/omrhashtable.h:68]: This does not belong in OMR.

File: include_core/omrhashtable.h [hunk 2]
```diff
@@ -129,6 +134,7 @@ typedef struct J9HashTableState {
 	void **pointerToCurrentNode;
 	uintptr_t iterateState;
 	struct J9PoolState poolState;
+	uintptr_t iterFlags;
```
Comment (gacholio) [inline include_core/omrhashtable.h:137]: I would just call this flags.

File: include_core/omrhashtable.h (hunk 1 above)
Comment (hangshao0) [reply include_core/omrhashtable.h]: Removed.

File: include_core/omrhashtable.h (hunk 2 above)
Comment (hangshao0) [reply include_core/omrhashtable.h]: Changed to flags.

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by hangshao0** [comment]
There are some builds failed due to:

```
ERROR: Error fetching remote repo 'origin'
hudson.plugins.git.GitException: Failed to fetch from https://github.com/eclipse/omr.git
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:915)
```

Others have passed. Not sure if you want to re-test @rwy0717.

**Comment by rwy0717** [comment]
@genie-omr build xlinux plinux aix win

**Comment by hangshao0** [comment]
Can this be merged ? This is blocking things on the OpenJ9 side.

**Comment by hangshao0** [comment]
@youngar Can you merge this ?

**Comment by DanHeidinga** [comment]
@mstoodle @rwy0717 We need this change for the Java 15 builds which code split on Aug 16.  Any chance this can be merged today to unblock the dependent OpenJ9 PRs?

**Comment by DanHeidinga** [comment]
The only build failure on linux_x86 is due to a github exception checking out the code.  No functional problem found on any other platform which suggests this should be fine to merge without a rebuild

**Comment by youngar** [comment]
LGTM

**Pull request merged**